### PR TITLE
EWS: alerts and reminders tests

### PR DESCRIPTION
**Pull request by kkrampa**

### CODE REVIEW

**Comment by kkrampa** [comment]
@czue It should be easiest to read commit by commit

**Comment by modonnell729** [comment]
@czue Is there anything else that still needs to be done before this can be merged?

**Comment by czue** [comment]
@kkrampa sorry no i haven't reviewed this yet. it is also not mergeable right now

**Comment by kkrampa** [comment]
@czue Updated

**Comment by czue** [comment]
only one more piece of feedback then good to go

File: custom/ewsghana/tests/test_notifications.py [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+    def test_all_facility_stockout(self):
+        """Send a notification because all facilities are stocked out of a product."""
+        assign_products_to_location(self.facility, [self.product])
+        assign_products_to_location(self.other_facility, [self.product])
+        assign_products_to_location(self.last_facility, [self.product])
+
+        create_stock_report(self.facility, {'tp': 0})
+        create_stock_report(self.other_facility, {'tp': 0})
+        create_stock_report(self.last_facility, {'tp': 0})
+
```
Comment (dimagimon) [inline custom/ewsghana/tests/test_notifications.py]: W293 blank line contains whitespace

File: custom/ewsghana/tests/test_notifications.py [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+            home_loc=self.region, program_id=program.get_id
+        )
+
+        assign_products_to_location(self.facility, [self.product])
+        assign_products_to_location(self.other_facility, [self.product])
+        assign_products_to_location(self.last_facility, [self.product])
+
+        generated = list(UrgentNonReporting(self.TEST_DOMAIN).get_notifications())
+        self.assertEqual(len(generated), 2)
+        self.assertEqual({generated[0].user.get_id, generated[1].user.get_id}, {self.user.get_id, other_user.get_id})
```
Comment (dimagimon) [inline custom/ewsghana/tests/test_notifications.py]: E501 line too long (117 > 115 characters)

File: custom/ewsghana/tests/test_notifications.py [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+        with mock.patch('custom.ewsghana.alerts.alert.send_sms_to_verified_number') as send:
+            self.notification.send()
+            self.assertFalse(send.called)
+
+    def test_opt_out(self):
+        """No message will be sent if the user has opted out of the notifications."""
+        self.user.user_data['sms_notifications'] = False
+        with mock.patch('custom.ewsghana.alerts.alert.send_sms_to_verified_number') as send:
+            self.notification.send()
+            self.assertFalse(send.called)
```
Comment (dimagimon) [inline custom/ewsghana/tests/test_notifications.py]: W292 no newline at end of file

**Comment by kkrampa** [comment]
@czue Updated

**Pull request merged**